src/schedule/zephyr_domain.c

```diff
@@ -70,11 +71,16 @@ static void zephyr_domain_thread_fn(void *p1, void *p2, void *p3)
 static void zephyr_domain_timer_fn(struct k_timer *timer)
 {
 	struct zephyr_domain *zephyr_domain = k_timer_user_data_get(timer);
+	uint64_t now = platform_timer_get(NULL);
 	int core;
 
 	if (!zephyr_domain)
 		return;
 
+	/* This loop should only run once */
+	while (zephyr_domain->ll_domain->next_tick < now)
+		zephyr_domain->ll_domain->next_tick += LL_TIMER_PERIOD_TICKS;
```

Comment:
Not sure if we need to finetune it now (if next_tick is removed later), but this looks a bit dangerous. Can next_tick be zero and this end up spinning for a longer time?